server-spi-private/src/main/java/org/keycloak/authentication/AuthenticationFlowException.java

```diff
@@ -76,4 +83,12 @@ public Response getResponse() {
     public List<AuthenticationFlowException> getAfeList() {
         return afeList;
     }
+
+    public String getUserErrorMessage() {
+        return userErrorMessage;
```

Comment:
the `userErrorMessage` is returned here, but never initialized. So this will always return null unless I am missing something? It seems something will also need to be added to the constructor IMO?